src/Core/Container/ContainerMaker.php

```diff
@@ -6,15 +6,30 @@
 use Dealroadshow\K8S\Data\Container;
 use Dealroadshow\K8S\Framework\App\AppInterface;
 use Dealroadshow\K8S\Framework\Core\Container\Env\EnvConfigurator;
+use Dealroadshow\K8S\Framework\Core\Container\Image\Image;
 use Dealroadshow\K8S\Framework\Core\Container\Lifecycle\LifecycleConfigurator;
 use Dealroadshow\K8S\Framework\Core\Container\Lifecycle\Probes\ProbesConfigurator;
 use Dealroadshow\K8S\Framework\Core\Container\Ports\PortsConfigurator;
 use Dealroadshow\K8S\Framework\Core\Container\Resources\ResourcesConfigurator;
 use Dealroadshow\K8S\Framework\Core\Container\Security\SecurityContextConfigurator;
 use Dealroadshow\K8S\Framework\Core\Container\VolumeMount\VolumeMountsConfigurator;
+use Dealroadshow\K8S\Framework\Middleware\ContainerImageMiddlewareInterface;
 
 class ContainerMaker
 {
+    /**
+     * @var ContainerImageMiddlewareInterface[]|iterable
+     */
+    private iterable $middlewares;
+
+    /**
+     * @param ContainerImageMiddlewareInterface[] $middlewares
+     */
+    public function __construct(iterable $middlewares)
+    {
+        $this->middlewares = $this->sortByPriority($middlewares);
```

Comment:
you can sort middlewares by priority in config. This is the feature of Symfony DI